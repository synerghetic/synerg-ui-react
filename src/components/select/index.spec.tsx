import React from 'react'
import Select from '.'
import { mount } from 'enzyme'

describe('Select', () => {
    const options = ['Hello', 'World']

    it('should render', () => {
        const component = mount(
            <Select onChange={() => void 0} options={options} />,
        )

        const html = component.html()
        expect(html).toContain('select')

        for (let i = 0; i < options.length; i++) {
            expect(html).toContain(options[i])
        }

        expect(component.html()).toMatchSnapshot()
    })

    it('should call onChange and update the variable', () => {
        let value = ''
        const onChange = (newValue: string) => {
            value = newValue
        }

        const component = mount(
            <Select onChange={onChange} options={options} />,
        )

        component
            .find('select')
            .simulate('change', { target: { value: options[0] } })

        expect(value).toBe(options[0])
    })

    it('should have an error', () => {
        const error = 'Oh no D:'

        const component = mount(
            <Select onChange={() => void 0} error={error} options={options} />,
        )

        const html = component.html()

        expect(html).toMatchSnapshot()

        expect(component).toContainMatchingElement('.error-message')
        expect(html).toContain(error)
    })
})
