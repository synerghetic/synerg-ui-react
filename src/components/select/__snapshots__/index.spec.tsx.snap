// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should have an error 1`] = `"<div class=\\"flex:column\\"><div class=\\"relative\\"><select name=\\"aze\\" style=\\"width: 100%;\\" class=\\"select input--error\\"><option>Hello</option><option>World</option></select><svg width=\\"24\\" height=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"#000\\" class=\\"input-icon\\"><path d=\\"M12 16a1 1 0 0 1-.64-.23l-6-5a1 1 0 1 1 1.28-1.54L12 13.71l5.36-4.32a1 1 0 0 1 1.41.15 1 1 0 0 1-.14 1.46l-6 4.83A1 1 0 0 1 12 16z\\"></path></svg></div><span class=\\"error-message \\">Oh no D:</span></div>"`;

exports[`Select should render 1`] = `"<div class=\\"flex:column\\"><div class=\\"relative\\"><select name=\\"aze\\" style=\\"width: 100%;\\" class=\\"select input\\"><option>Hello</option><option>World</option></select><svg width=\\"24\\" height=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"#000\\" class=\\"input-icon\\"><path d=\\"M12 16a1 1 0 0 1-.64-.23l-6-5a1 1 0 1 1 1.28-1.54L12 13.71l5.36-4.32a1 1 0 0 1 1.41.15 1 1 0 0 1-.14 1.46l-6 4.83A1 1 0 0 1 12 16z\\"></path></svg></div><span class=\\"error-message hidden\\"></span></div>"`;
